### DISCUSSION TITLE
NIC Speed Rating Module

### DISCUSSION THREAD

**Discussion by jrstrunk** [fastfetch-cli/fastfetch#1208] [Ideas]
I don't have the know how to it myself, but it would be cool to have a module for displaying the NICs on a system and their rated speeds, so that people can flex their 2.5Gb and 10Gb capable builds 🔥

I know there is a network IO module, but this takes a while to load and does not show what the rated speed is

This could also be a formatting option for the netio module :)

**Comment by nth10sd** [comment]
@CarterLi While #1209 added support for this via `fastfetch --format json -s localip --localip-default-route-only false`, the speed (e.g. 10000) still doesn't seem to show up even with `fastfetch -c all.jsonc`.

The [schema](https://github.com/fastfetch-cli/fastfetch/blob/74fcd8b1c31421f481d310308e50b8dad0ccbc57/doc/json_schema.json#L1619) does not seem to have a `showSpeed` boolean setting, which could be added to the default output or at least to `all.jsonc`, to show `10 Gbps`.

As a comparison, `neowofetch` via `hyfetch` shows `Network: 10 Gbps` when run on my computer on their default output.

Let me know if I'm making wrong assumptions here.
